FRAuth/FRAuth/Callbacks/DeviceBindingCallback.swift

```diff
@@ -0,0 +1,221 @@
+            else if let outputName = output[CBConstants.name] as? String, outputName == CBConstants.title, let outputValue = output[CBConstants.value] as? String {
+                title = outputValue
+            }
+            else if let outputName = output[CBConstants.name] as? String, outputName == CBConstants.subtitle, let outputValue = output[CBConstants.value] as? String {
+                subtitle = outputValue
+            }
+            else if let outputName = output[CBConstants.name] as? String, outputName == CBConstants.description, let outputValue = output[CBConstants.value] as? String {
+                promptDescription = outputValue
+            }
+            else if let outputName = output[CBConstants.name] as? String, outputName == CBConstants.timeout, let outputValue = output[CBConstants.value] as? Int {
+                timeout = outputValue
+            }
+        }
+        
+        guard userId != nil, userName != nil, challenge != nil, deviceBindingAuthenticationType != nil, title != nil, subtitle != nil, promptDescription != nil else {
+            throw AuthError.invalidCallbackResponse("Missing an output value")
+        }
+        
+        for input in inputs {
+            if let name = input[CBConstants.name] as? String, name.contains(CBConstants.jws) {
+                self.jwsKey = name
+            }
+            else if let name = input[CBConstants.name] as? String, name.contains(CBConstants.deviceName) {
+                self.deviceNameKey = name
+            }
+            else if let name = input[CBConstants.name] as? String, name.contains(CBConstants.deviceId) {
+                self.deviceIdKey = name
+            }
+            else if let name = input[CBConstants.name] as? String, name.contains(CBConstants.clientError) {
+                self.clientErrorKey = name
+            }
+        }
+        
+        guard jwsKey != nil, deviceNameKey != nil, deviceIdKey != nil, clientErrorKey != nil else {
+            throw AuthError.invalidCallbackResponse("Missing an input value")
+        }
+        
+        try super.init(json: json)
+        
+        self.type = callbackType
+        self.response = json
+    }
+    
+    
+    /// Bind the device.
+    /// - Parameter completion Completion block for Device binding result callback
+    func bind(completion: @escaping DeviceBindingResultCallback) {
+        execute(authInterface: nil, deviceId: nil, completion)
+    }
+    
+    
+    /// Helper method to execute binding , signing, show biometric prompt.
+    /// - Parameter authInterface: Interface to find the Authentication Type - procide nil to default to getDeviceBindingAuthenticator()
+    /// - Parameter deviceId: Interface to find the Authentication Type - provide nil to default to FRDevice.currentDevice?.identifier.getIdentifier()
+    /// - Parameter completion: completion Completion block for Device binding result callback
+    open func execute(authInterface: DeviceAuthenticator?,
```

Comment:
i am not sure we need open,  protected open is a standard in kotlin to override and method is not exposed , in swift open method can be called by anyone i believe . so i am kind of thinking this

`class DeviceBindingCallback {
    fileprivate func execute() {
        print("Foo")
    }
}

class Bar: DeviceBindingCallback {
    fileprivate override func jey() {
        print("Bar")
    }
}`

Reply:
so i dont want this 

let callback = DeviceBindCallBack()
callback.execute()

but i want this

class GeorgeCallback: DeviceBindCallBack {
   override func execute()
}